modules/apps/notification/notification-web/src/main/resources/META-INF/resources/js/components/SettingsContainer/SecondaryRecipients.tsx

```diff
@@ -41,6 +42,49 @@ export function SecondaryRecipient({
 	const [ccRolesList, setCCRolesList] = useState<MultiSelectItem[]>([]);
 	const [recipient] = values.recipients as EmailRecipients[];
 
+	const handleRecipientRoleChange = (
+		items: MultiSelectItem[],
+		recipientKey: 'cc' | 'bcc',
+		setRoleList: (value: MultiSelectItem[]) => void
+	) => {
+		const newRecipients = handleMultiSelectRoleItemsChange(items);
+
+		setValues({
+			...values,
+			recipients: [
+				{
+					...(values.recipients[0] as EmailRecipients),
+					[recipientKey]: newRecipients,
+				},
+			],
+		});
+
+		setRoleList(items);
+	};
+
+	const handleRecipientTypeChange = (
+		newRecipientTypeValue: string,
+		recipientKey: 'cc' | 'bcc',
+		roleList: MultiSelectItem[],
+		recipientTypeKey: 'ccType' | 'bccType',
+		setRoleList: (value: MultiSelectItem[]) => void
+	) => {
+		if (newRecipientTypeValue === 'email') {
+			const newRoleList = uncheckMultiSelectItemChildrens(roleList);
+			setRoleList(newRoleList);
+		}
+		setValues({
+			...values,
+			recipients: [
+				{
+					...recipient,
+					[recipientKey]: newRecipientTypeValue === 'email' ? '' : [],
```

Comment:
Hey @vngoliveira, I realized that this modification is the main change to fix the bug:

`newRecipientTypeValue === 'email' ? '' : []`

But we are not creating a test for this function, thus ensuring that this bug does not occur again.

To unit test this function in a simple way, I have two suggestions:

1. The first would be to return the content that is being used in `setValues`. This means that we always need to call `setValues` ​​after calling this function. And then in the unit test, we can guarantee that the return of the function is as expected. Similar to the test created for the function that clears multipleselect items

2. Isolate the condition: `newRecipientTypeValue === 'email' ? '' : []` In another function, test this function unitarily.

In the first approach, it will be necessary to mock the data structure by simulating `values`. We will be ensuring that the return of `recipientKey` and `recipientTypeKey` will be as expected.

On the other hand, the second test will be simpler, testing the condition causing the bug in isolation. No need to move `setValues` ​​out of the function.